### PR TITLE
Development

### PR DESCRIPTION
**Pull request by PetarHarambasa**

### CODE REVIEW

File: src/hr/algebra/pi/factory/BigVehicleFactory.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package hr.algebra.pi.factory;
+
+import hr.algebra.pi.model.BigBus;
+import hr.algebra.pi.model.BigTruck;
+import hr.algebra.pi.model.IVehicle;
+
+public class BigVehicleFactory extends VehicleFactory{
```
Comment (bornaskracic) [inline src/hr/algebra/pi/factory/BigVehicleFactory.java:7]: Interesting combination of simple factory method and regular factory pattern!

File: src/hr/algebra/pi/model/Terminal.java [hunk 2]
```diff
@@ -1,81 +1,94 @@
...
-        if (vehicle.size.equals("small")) {
-            if (smallTrain.addVehicle(vehicle)) {
-                if (vehicle.type.equals("car")) {
-                    ticketPrice = TicketPrices.SMALL_VEHICLE_CAR_PRICE;
-                } else if (vehicle.type.equals("van")) {
-                    ticketPrice = TicketPrices.SMALL_VEHICLE_VAN_PRICE;
-                }
+    public boolean parkVehicle(String type, String size, double gasLevel, double batteryLevel) {
+        IVehicle vehicle = null;
+        if (size.equalsIgnoreCase("small")) {
```
Comment (bornaskracic) [inline src/hr/algebra/pi/model/Terminal.java:25]: Be careful with long if statements, especially if they depend on the object type :)

File: src/hr/algebra/pi/model/Employee.java [hunk 3]
```diff
@@ -1,23 +1,13 @@
 package hr.algebra.pi.model;
 
 public class Employee {
```
Comment (bornaskracic) [inline src/hr/algebra/pi/model/Employee.java:3]: Hint: Maybe Employee should be responsible for parking and refilling vehicles?

File: src/hr/algebra/pi/test/TerminalTest.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+
+public class TerminalTest {
+    Terminal terminal = new Terminal();
+
+    @Test
+    public void testParkSmallCarSuccess() {
```
Comment (bornaskracic) [inline src/hr/algebra/pi/test/TerminalTest.java:13]: Be careful when unit testing, you usually test one functionality and side-effect, especially if we are talking about 'by the books' examples :)

**Review by bornaskracic — APPROVED** [review]

File: src/hr/algebra/pi/model/Employee.java (hunk 3 above)
Comment (PetarHarambasa) [reply src/hr/algebra/pi/model/Employee.java:3]: Agree, i will fix that.

**Pull request merged**